proxy/src/console/provider.rs

```diff
@@ -441,10 +442,22 @@ pub struct ApiLocks {
     node_locks: DashMap<EndpointCacheKey, Arc<Semaphore>>,
     permits: usize,
     timeout: Duration,
-    registered: prometheus::IntCounter,
-    unregistered: prometheus::IntCounter,
-    reclamation_lag: prometheus::Histogram,
-    lock_acquire_lag: prometheus::Histogram,
+    metrics: &'static ApiLockMetrics,
+}
+
+#[derive(MetricGroup)]
```

Comment:
I don't really like that we define metric here. What do you think about moving it to `metrics.rs`?

Reply:
Sure. The intention was to keep it close to the original definition, but I don't mind moving it.